src/ruby/lib/grpc/generic/active_call.rb

```diff
@@ -582,6 +584,23 @@ def attach_peer_cert(peer_cert)
 
     private
 
+    # response_status_as_trailers_only sends a status as Trailers-only
+    # https://github.com/grpc/grpc/blob/ceecf80283e6ca184df587f76ed053f1f5295b7f/doc/PROTOCOL-HTTP2.md
+    def response_status_as_trailers_only(code, details, metadata: {})
+      @send_initial_md_mutex.synchronize do
+        # Can't send Trailers-Only since Response-Headers have already been sent
+        return false if @metadata_sent
+        @metadata_sent = true
+      end
+
```

Comment:
After my comment above, @ganmacs can you please also add a comment here that we are attempting to send initial metadata in the same op batch as trailing metadata in order to experimentally support retries?

Reply:
https://github.com/grpc/grpc/pull/16490/commits/3f9fb2da2e009e6197362c30f7ecd1d104daafe3 added.